src/sentry/static/sentry/app/views/usageStats/usageChart/utils.tsx

```diff
@@ -1,26 +1,40 @@
 import moment from 'moment';
 
-import {DataCategory} from 'app/types';
+import {DataCategory, IntervalPeriod} from 'app/types';
 
 import {formatUsageWithUnits} from '../utils';
 
-export function getDateFromMoment(m: moment.Moment) {
-  return m.format('MMM D');
+export function getDateFromMoment(m: moment.Moment, interval: IntervalPeriod = '1d') {
+  const unit = interval.replace(/[0-9]/g, '');
```

Comment:
Similar to above, would it be better to base this on the length of time instead of the unit?